graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/pivot/buckets/TimeUnitInterval.java

```diff
@@ -24,21 +24,45 @@
 import org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramInterval;
 import org.graylog2.plugin.indexer.searches.timeranges.TimeRange;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
 @AutoValue
 @JsonTypeName(TimeUnitInterval.type)
 @JsonDeserialize(builder = TimeUnitInterval.Builder.class)
 public abstract class TimeUnitInterval implements Interval {
     public static final String type = "timeunit";
+    public static final Pattern TIMEUNIT_PATTERN = Pattern.compile("(?<quantity>\\d+)(?<unit>[smhdwM])");
 
     @JsonProperty
     public abstract String type();
 
     @JsonProperty
     public abstract String timeunit();
 
+    private String adjustUnitsLongerThanDays(String timeunit) {
+        final Matcher matcher = TIMEUNIT_PATTERN.matcher(timeunit());
+        checkArgument(matcher.matches(),
+                "Time unit must be {quantity}{unit}, where quantity is a positive number and unit [smhdwM].");
+        final int quantity = Integer.parseInt(matcher.group("quantity"), 10);
```

Comment:
What's the benefit of specifying a radix of 10 here explicitly? I would find it clearer, if we omitted it.

Reply:
No other reason besides being as explicit as possible. If it's too explicit, I will remove it :)